src/app/(sidebar)/smart-contracts/contract-explorer/components/InvokeContractForm.tsx

```diff
@@ -144,14 +144,26 @@ export const InvokeContractForm = ({
 
     if (walletKit?.publicKey) {
       try {
-        const result = await walletKitInstance.walletKit.signTransaction(
+        // Add timeout to prevent endless loading
+        // when user exits out of the extension
+        const timeoutPromise = new Promise<never>((_, reject) => {
+          setTimeout(() => {
+            reject(
+              new Error("Transaction signing timed out. Please try again."),
+            );
+          }, 20000);
```

Comment:
We need to make sure we clear the timeout. We can set a variable outside of the `signTx()` method (`let timeout = null`, for example). And then clear it on line 139 whenever `signTx()` method fires. And then:

```suggestion
          timeout = setTimeout(() => {
            reject(
              new Error("Transaction signing timed out. Please try again."),
            );
            clearTimeout(timeout)
          }, 20000);
```

Reply:
I'll set the timeout variable within `signTx` so that in case of being clicked multiple times, it won't use the same `timeout` variable